### ISSUE TITLE
High memory usage

### ISSUE THREAD

**Issue by Snack32** [jlesage/docker-filebot#57]
Hello,

First of all, I like to express my gratitude for your work (jdownloader, filebot...) and the trully masterpiece and crown jewel, your baseimage.

However, I notice a strange behavour in your filebot docker image, specifically a huge consumption of memory after only 10-15 renames.
My system have 16GB of RAM and filebot is, by far, the most expensive of them, consuming after the aforementioned 10-15 renames, 1,8GB or 2GB that never released.

The strange thing is that it is not the filebot itself that consumes it (it reports only barely 51MB used memory), so it must be something else.
Ie,:

Filebot own report (docker exec filebot /opt/filebot/filebot -script fn:sysinfo):
FileBot 4.9.3 (r8340) JDK8
JNA Native: 5.2.2
MediaInfo: 21.03
Tools: fpcalc/1.4.3 p7zip/16.02 unrar/5.61
Extended Attributes: OK
Unicode Filesystem: OK
Script Bundle: 2021-05-30 (r751)
Groovy: 3.0.7
JRE: OpenJDK Runtime Environment 1.8.0_275
JVM: 64-bit OpenJDK 64-Bit Server VM
CPU/MEM: 4 Core / 3.7 GB Max Memory / 51 MB Used Memory
OS: Linux (amd64)
HW: Linux caeab5b1fab9 5.9.0-0.bpo.5-amd64 #1 SMP Debian 5.9.15-1~bpo10+1 (2020-12-31) x86_64 GNU/Linux
CPU/MEM: Intel(R) Pentium(R) Silver J5040 CPU @ 2.00GHz [MemTotal: 16 GB | MemFree: 10 GB | MemAvailable: 11 GB | SwapT                    otal: 1.0 GB | SwapFree: 1.0 GB]
STORAGE: overlay [/] @ 433 GB | fuse.mergerfs [/storage] @ 1,126 TB | ext4 [/config] @ 433 GB | ext4 [/output] @ 433 GB                     | ext4 [/watch] @ 433 GB
USER: root
DATA: /config
Package: DOCKER
License: FileBot License *edited* (Valid-Until: 2022-06-11)
Done ヾ(＠⌒ー⌒＠)ノ

Docker stats (docker stats filebot --no-stream):
CONTAINER ID   NAME      CPU %     MEM USAGE / LIMIT    MEM %     NET I/O          BLOCK I/O        PIDS
caeab5b1fab9   filebot   0.67%     1.816GiB / 15.3GiB   11.87%    7.73MB / 8.9MB   73.1MB / 766kB   68

Memory stat (from /sys/fs/cgroup/memory/docker/$FilebotDockerID/memory.stat):
cache 44064768
rss 1914183680
rss_huge 1799356416
shmem 7028736
mapped_file 7569408
dirty 0
writeback 0
swap 0
pgpgin 1101144
pgpgout 1067211
pgfault 1151139
pgmajfault 297
inactive_anon 1921421312
active_anon 135168
inactive_file 17244160
active_file 19677184
unevictable 0
hierarchical_memory_limit 9223372036854771712
hierarchical_memsw_limit 9223372036854771712
total_cache 44064768
total_rss 1914183680
total_rss_huge 1799356416
total_shmem 7028736
total_mapped_file 7569408
total_dirty 0
total_writeback 0
total_swap 0
total_pgpgin 1101144
total_pgpgout 1067211
total_pgfault 1151139
total_pgmajfault 297
total_inactive_anon 1921421312
total_active_anon 135168
total_inactive_file 17244160
total_active_file 19677184
total_unevictable 0

Do you know why this happens? Or should it be considered normal?

In any case, thanks for your work!

**Comment by jlesage** [comment]
Can you provide the output of the following command:
```
docker exec <container name> ps -o pid,rss,comm,args
```

**Comment by Snack32** [comment]
Of course. Filebot docker restart and analyzing (without rename) only two files (1 GiB memory):

root@NAS:~# docker exec filebot ps -o pid,rss,comm,args
PID   RSS  COMMAND          COMMAND
    1    4 s6-svscan        s6-svscan -s -t0 /var/run/s6/services
   35    4 s6-supervise     s6-supervise s6-fdholderd
  642    4 s6-supervise     s6-supervise xvfb
  644    4 s6-supervise     s6-supervise nginx
  648    4 s6-supervise     s6-supervise amc
  649    4 s6-supervise     s6-supervise x11vnc
  650    4 s6-supervise     s6-supervise openbox
  651    4 s6-supervise     s6-supervise app
  791    4 s6-fdholderd     s6-fdholderd -1 -i rules
  806  34m Xvfb             /usr/bin/Xvfb :0 -screen 0 1280x768x24
  842  11m openbox          /usr/bin/openbox
  871 4216 nginx            nginx: master process /usr/sbin/nginx -c /etc/nginx/nginx.conf -g daemon off;
  896  15m x11vnc           /usr/bin/x11vnc -display :0 -rfbport 5900 -rfbportv6 -1 -no6 -noipv6 -httpportv6 -1 -forever -desktop FileBot -cursor arrow -shared -nopw
  906 1584 bash             bash /run/s6/services/amc/amc
  916 1808 nginx            nginx: worker process
  918 1.0g java             /usr/lib/jvm/java-1.8-openjdk/bin/java -Dapplication.deployment=docker -Dnet.filebot.license=/config/license.psm -Dnet.filebot.media.parser=libmediainfo -Dnet.filebot.archive.extractor=ShellExecutables -Dunixfs=false -DuseExtendedFileAttributes=true -DuseCreationDate=false -Djava.net.useSystemProxies=false -Djna.nosys=false -Djna.nounpack=true -Djna.boot.library.path=/usr/lib -Djna.library.path=/usr/lib -Djava.library.path=/usr/lib -Dapplication.dir=/config -Dapplication.cache=/config/cache -Djava.io.tmpdir=/config/tmp -Dfile.encoding=UTF-8 -Dsun.jnu.encoding=UTF-8 -Duser.home=/config -Djava.util.prefs.PreferencesFactory=net.filebot.util.prefs.FilePreferencesFactory -Dnet.filebot.util.prefs.file=/config/prefs.properties -DuseGVFS=false -Xbootclasspath/p:/usr/share/java/jna.jar -Dapplication.update=skip -Dprism.verbose=true -Dprism.order=sw -Dawt.useSystemAAFontSettings=on -Djdk.gtk.version=2 -jar /opt/filebot/jar/filebot.jar
  919 1808 nginx            nginx: worker process
 1075  508 sleep            sleep 1800
 1174    4 ps               ps -o pid,rss,comm,args

**Comment by Snack32** [comment]
Partially solved. 

At the end, I think it's all due to how java manages memory (max heap, etc.).

So, as I have seen that JAVA_OPTS variable exists in your filebot starting script (but not used), I simply added to the compose the Xmx environment variable with 512m.
(...)
environment:
  USER_ID: 1000
  GROUP_ID: 1000
  JAVA_OPTS: -Xmx512m

With this, I have managed to keep the memory usage around 600-650M:
 882 649m java /usr/lib/jvm/java-1.8-openjdk/bin/java -Dapplication.deployment=docker (...)

I know it could have been done by limiting memory usage in the container itself, but I notice that it doesn't run as smoothly as limiting java.

I write it here in case someone in the future has the same problem.

**Comment by jlesage** [comment]
Thanks for the update!
